### PR TITLE
Using KyberDao to deploy KyberStaking contract

### PR DESCRIPTION
**Pull request by manhlx3006**

### CODE REVIEW

File: contracts/sol6/Dao/KyberDAO.sol [hunk 1]
```diff
@@ -139,6 +132,15 @@ contract KyberDAO is IKyberDAO, EpochUtils, ReentrancyGuard, Utils5, DaoOperator
...
         });
+
+        // deploy staking contract 
+        staking = new KyberStaking({
+            _kncToken: _knc,
+            _epochPeriod: _epochPeriod,
+            _startTimestamp: _startTimestamp,
+            _daoContract: address(this)
+        });
+        emit KyberStakingDeployed(address(staking));
```
Comment (ilanDoron) [inline contracts/sol6/Dao/KyberDAO.sol:143]: Don't think an event is required here.
anyway the address can be easily fetched. 
And will never have another one

File: contracts/sol6/Dao/mock/MockKyberDaoMoreGetters.sol [hunk 2]
```diff
@@ -33,7 +31,7 @@ contract MockKyberDaoMoreGetters is KyberDAO {
     }
 
     function replaceStakingContract(address _staking) public {
```
Comment (ilanDoron) [inline contracts/sol6/Dao/mock/MockKyberDaoMoreGetters.sol:33]: function name not clear.
maybe fetchStakingCotractAddress

Comment (manhlx3006) [reply contracts/sol6/Dao/mock/MockKyberDaoMoreGetters.sol:33]: oh it is for changing staking contract to be a normal address, so we can call some functions in DAO that require sender is staking.

File: contracts/sol6/Dao/KyberDAO.sol (hunk 1 above)
Comment (manhlx3006) [reply contracts/sol6/Dao/KyberDAO.sol:143]: ok, can remove event, staking is public getter anyway

File: contracts/sol6/Dao/KyberStaking.sol [hunk 3]
```diff
@@ -39,47 +37,17 @@ contract KyberStaking is IKyberStaking, EpochUtils, ReentrancyGuard {
         address _kncToken,
         uint256 _epochPeriod,
         uint256 _startTimestamp,
-        address _daoContractSetter
```
Comment (ilanDoron) [inline contracts/sol6/Dao/KyberStaking.sol:42]: we can directly take this parameters as IKyberDAO
and above for knc
IERC20
And avoid the casting later.

 I think this is what we usually do.

File: contracts/sol6/Dao/mock/MockKyberDaoMoreGetters.sol (hunk 2 above)
Comment (ilanDoron) [reply contracts/sol6/Dao/mock/MockKyberDaoMoreGetters.sol:33]: Oh
OK

**Pull request merged**